syncback/Tiltfile

```diff
@@ -30,11 +36,22 @@ def syncback(name, k8s_object, src_dir, paths=None, target_dir='.', container=''
     """
 
     incl_excl = ''
+    if ignore == None:
+        ignore = []
+
     if paths:
         # TODO: if you're rsync-savvy you might want to do the wildcarding manually--
         #   give an option to turn off automatic +'***'
         includes = ' '.join(['--include="{}***"'.format(p) for p in paths])
         incl_excl = '{} --exclude="*"'.format(includes)
+    else:
+        # Sync the entire src_dir. Danger, Will Robinson! Ignore some stuff
+        # that probably exists locally but not in your container so it
+        # doesn't get wiped out locally on your first sync
+        ignore.extend(DEFAULT_IGNORES)
```

Comment:
it's generally considered bad practice to modify your inputs. for example, if you have

```
shared_ignores = []
syncback(..., ignore = shared_ignores)
syncback(..., ignore = shared_ignores)
```

you will end up modifying it twice.

Reply:
whooops I guess I assumed that starlark would guard against that. Will change tomorrow.